### PR TITLE
Update the supporter data dynamo table when performing product switch

### PR DESCRIPTION
**Pull request by jonflynng**
## What does this change?

 This updates the [Supporter Data table in Dynamo](https://eu-west-1.console.aws.amazon.com/dynamodbv2/home?region=eu-west-1#item-explorer?table=SupporterProductData-PROD) with the latest product data when performing a switch. The table grants the user the appropriate benefits when viewing the app. This is necessary to perform during the switch as the user will be re-directed to the app after the switch and expect to get past the pay wall.

### CODE REVIEW

**Comment by jonflynng** [comment]
[Update the supporter product data datastore when the switch is complete](https://trello.com/c/AhTXA2qS/3756-update-the-supporter-product-data-datastore-when-the-switch-is-complete)

File: handlers/product-move-api/src/main/scala/com/gu/productmove/endpoint/move/ProductMoveEndpoint.scala [hunk 1]
```diff
@@ -221,7 +228,6 @@ object ProductMoveEndpoint {
           account.basicInfo.IdentityId__c,
         ),
       )
-      .addLogMessage("SQS sendEmail()")
```
Comment (jonflynng) [inline handlers/product-move-api/src/main/scala/com/gu/productmove/endpoint/move/ProductMoveEndpoint.scala:224]: Removed adding an extra string to the logging here as the SQS service has explicit logging on each function.

File: handlers/product-move-api/src/main/scala/com/gu/productmove/endpoint/move/ProductMoveEndpoint.scala [hunk 2]
```diff
@@ -242,13 +249,28 @@ object ProductMoveEndpoint {
...
         ZIO.succeed(()).fork
 
-    sqsRequests = emailFuture.zip(refundFuture).zip(salesforceTrackingFuture)
-    _ <- sqsRequests.join
+    supporterPlusRatePlanIds <- ZIO.fromEither(getSupporterPlusRatePlanIds(stage, ratePlanCharge.billingPeriod))
+    amendSupporterProductDynamoTableFuture <- Dynamo
+      .writeItem(
+        SupporterRatePlanItem(
+          subscriptionName,
+          identityId = account.basicInfo.IdentityId__c.getOrElse(""),
```
Comment (rupertbates) [inline handlers/product-move-api/src/main/scala/com/gu/productmove/endpoint/move/ProductMoveEndpoint.scala:261]: If we don't have an identity id we should fail at this point a record with one is useless

Comment (jonflynng) [reply handlers/product-move-api/src/main/scala/com/gu/productmove/endpoint/move/ProductMoveEndpoint.scala:261]: Should we fail here or further up, before doing the rate plan update?

Comment (jonflynng) [reply handlers/product-move-api/src/main/scala/com/gu/productmove/endpoint/move/ProductMoveEndpoint.scala]: am handling it [here](https://github.com/guardian/support-service-lambdas/pull/1823/files#diff-8279c56bbcc6ef4ba4156a1092ef21917f961403ecd46fd162cbea1b190688b5R200-R205). Have also updated tests to have an identityId in the account response and added a test if identityId is null.

File: handlers/product-move-api/src/main/scala/com/gu/productmove/endpoint/move/ProductMoveEndpoint.scala [hunk 3]
```diff
@@ -242,13 +254,28 @@ object ProductMoveEndpoint {
...
+    supporterPlusRatePlanIds <- ZIO.fromEither(getSupporterPlusRatePlanIds(stage, ratePlanCharge.billingPeriod))
+    amendSupporterProductDynamoTableFuture <- Dynamo
+      .writeItem(
+        SupporterRatePlanItem(
+          subscriptionName,
+          identityId = identityId,
+          gifteeIdentityId = None,
+          productRatePlanId = supporterPlusRatePlanIds.ratePlanId,
+          productRatePlanName = "product-move-api added Supporter Plus Monthly",
+          termEndDate = todaysDate.plusYears(1),
```
[the remark this answers is not in the document]
Comment (jonflynng) [reply handlers/product-move-api/src/main/scala/com/gu/productmove/endpoint/move/ProductMoveEndpoint.scala:270]: ok

[the remark this answers is not in the document]
Comment (jonflynng) [reply handlers/product-move-api/src/main/scala/com/gu/productmove/endpoint/move/ProductMoveEndpoint.scala:270]:
```suggestion
          termEndDate = todaysDate.plusDays(7),
```

**Review by rupertbates — APPROVED** [review]
